### PR TITLE
test: more magic link flow integration tests

### PR DESCRIPTION
**Pull request by danielgent**
We only had the happiest of happy-paths for the magic link

### CODE REVIEW

File: integration-test/flows/magic-link-flow.spec.ts [hunk 1]
```diff
@@ -15,106 +16,354 @@ describe("code-flow", () => {
     worker.stop();
   });
 
-  it("should log in using the sent magic link", async () => {
```
Comment (danielgent) [inline integration-test/flows/magic-link-flow.spec.ts:18]: this was previously just the happiest of happy paths :wink:

File: integration-test/flows/magic-link-flow.spec.ts [hunk 2]
```diff
@@ -15,106 +16,354 @@ describe("code-flow", () => {
...
-    const nonce = "ehiIoMV7yJCNbSEpRq513IQgSX7XvvBM";
-    const redirect_uri = "https://login.example.com/sv/callback";
-    const response_type = "token id_token";
-    const scope = "openid profile email";
-    const state = "state";
+  describe("should log in using the sent magic link, when", () => {
+    it("is a new sign up", async () => {
```
Comment (danielgent) [inline integration-test/flows/magic-link-flow.spec.ts:20]: the code is potentially quite different when the user already exists vs a new signup. better to test both

File: integration-test/flows/magic-link-flow.spec.ts [hunk 3]
```diff
@@ -15,106 +16,354 @@ describe("code-flow", () => {
...
+      const response = await worker.fetch("/passwordless/start", {
+        headers: {
+          "content-type": "application/json",
+        },
+        method: "POST",
+        body: JSON.stringify({
+          authParams: AUTH_PARAMS,
+          client_id: "clientId",
+          connection: "email",
+          email: "new-user@example.com",
```
Comment (danielgent) [inline integration-test/flows/magic-link-flow.spec.ts:54]: if you have whitespace diffs turned off there shouldn't be _that_ much, but I've changed the email here to be more explicitly a new user, after first asserting via the mgmt-api that the user doesn't already exist

File: integration-test/flows/magic-link-flow.spec.ts [hunk 4]
```diff
@@ -15,106 +16,354 @@ describe("code-flow", () => {
...
+      expect(restOfIdTokenPayload).toEqual({
+        aud: "clientId",
+        name: "new-user@example.com",
+        email: "new-user@example.com",
+        email_verified: true,
+        nonce: "enljIoQjQQy7l4pCVutpw9mf001nahBC",
+        iss: "https://example.com/",
+      });
+    });
+    it("is an existing user", async () => {
```
Comment (danielgent) [inline integration-test/flows/magic-link-flow.spec.ts:133]: and now test when the user already exists

File: integration-test/flows/magic-link-flow.spec.ts [hunk 5]
```diff
@@ -15,106 +16,357 @@ describe("code-flow", () => {
...
+    expect(redirectUri2.pathname).toBe("/sv/expired-code");
+    expect(redirectUri2.searchParams.get("email")).toBe(
+      encodeURIComponent("another@email.com"),
+    );
   });
   // TO TEST
-  // - that can reuse same magic link again
-  // - that a "bad" magic link doesn't work?  Not sure how much time we want to spend on validating access protection... probably a bit for security holes
+  // like code-flow
+  // it("should return existing primary account when logging in with new code sign on with same email address", async () => {
```
Comment (danielgent) [inline integration-test/flows/magic-link-flow.spec.ts:371]: I think we should do this at some point :see_no_evil:

**Review by markusahlstrand — APPROVED** [review]
Looks good

**Pull request merged**